chinese_tutorial_rvest_httr.Rmd

```diff
@@ -0,0 +1,490 @@
+                         cat("Error:",conditionMessage(e),"\n")
+                         return(NULL)})
+}
+
+lapply(u,get_rate)
+```
+
+```R
+## Error: arguments imply differing number of rows: 1, 0
+```
+
+```R
+## [[1]]
+## [1] 9.1
+## 
+## [[2]]
+## [1] 8.9
+## 
+## [[3]]
+## NULL
+## 
+## [[4]]
+## [1] 9
+```
+
+
+
+### 案例三：抓取豆瓣电影 top250
+
+链接：https://movie.douban.com/top250
+
+要求：获取豆瓣电影 top250 列表
+
+```R
+get_top250 <- function(i,getLink=FALSE){
+        require(httr)
+        u <- sprintf("https://movie.douban.com/top250?start=%d&filter=",i)
+        page <- u %>% GET() %>% content
+        title <- page %>% html_nodes(".title:nth-child(1)") %>% html_text()
+        if(getLink){
+                detail_link <- page %>% html_nodes("div.hd > a") %>% html_attr("href")
+                df <- data.frame(title=title,
+                                 detail_link=detail_link,
+                                 stringsAsFactors = FALSE)
+                return(df)
+        }else return(title)
+}
+
+# 不需要得到电影详情页面链接
+top250 <- lapply(seq(0,225,by=25), get_top250)
+
+top250_vec <- top250 %>% unlist
+
+save(top250_vec, file="top250_vec.Rdata")
+
+# 需要得到电影详情页面链接
+top250_withLink <- lapply(seq(0,225,by=25),
+                          get_top250, getLink=TRUE)
+
+top250_df <- top250_withLink %>% Reduce(rbind,.)
+
+save(top250_df, file="top250_df.Rdata")
+```
+
+
+
+### 案例四：抓取豆瓣热门电影「动态页面的抓取」
+
+链接：https://movie.douban.com/
+
+要求：获取豆瓣电影首页展示出来的热门电影列表
+
+```R
+library(httr)
+library(jsonlite)
+
+u <- "https://movie.douban.com/j/search_subjects?type=movie&tag=热门&page_limit=40&page_start=0"
+
+## 一、发起 GET 请求获取数据
+headers <- c("User-Agent"="Mozilla/5.0 (Windows NT 10.0; WOW64; rv:51.0) Gecko/20100101 Firefox/51.0",
+             "X-Requested-With"="XMLHttpRequest",
+             "Cookie"='ll="108300"; bid=lVscia-_MWA; _pk_ref.100001.4cf6=%5B%22%22%2C%22%22%2C1499860968%2C%22https%3A%2F%2Fwww.baidu.com%2Fbaidu%3Fwd%3D%25E8%25B1%2586%25E7%2593%25A3%26tn%3Dmonline_4_dg%26ie%3Dutf-8%22%5D; _pk_id.100001.4cf6=c1f0706d81e95c0b.1488797655.52.1499860968.1499858546.; __utma=30149280.647508500.1488797657.1499856903.1499860968.52; __utmz=30149280.1499418495.32.17.utmcsr=douban.com|utmccn=(referral)|utmcmd=referral|utmcct=/; __utma=223695111.2135525097.1488797657.1499856903.1499860968.52; __utmz=223695111.1499860968.52.24.utmcsr=baidu|utmccn=(organic)|utmcmd=organic|utmctr=%E8%B1%86%E7%93%A3; _vwo_uuid_v2=E6655FD962C6D1103E3FF2A4F47436B5|9fe9945dedabcf10658c10bb20c79bd9; viewed="3283973"; gr_user_id=39d25c13-cc3d-4203-90b3-998f5d747d38; __yadk_uid=BYBkYyvVzLcrrN3BgQhaxvVdx4tJVLvt; ue="1329262214@qq.com"; __utmv=30149280.14318; ps=y; push_noty_num=0; push_doumail_num=0; ap=1; __utmc=30149280; __utmc=223695111; as="https://movie.douban.com/"; _pk_ses.100001.4cf6=*; __utmb=30149280.0.10.1499860968; __utmb=223695111.0.10.1499860968',
+             "Cache-Control"="max-age=0")
+
+r <- u %>% GET(add_headers(headers),verbose())
+
+bodyData <- r %>% content()
+
+### 1.第一种方法
```

Comment:
Hashtag headers should not be followed by numbers since the hashtags will create numbered subheadings

Reply:
These are in the r code, so I think it's ok to leave them like this.